### PR TITLE
Add pre- and post-commit hooks

### PR DESCRIPTION
**Pull request by nilsdeppe**

### CODE REVIEW

File: tools/Hooks/CheckFileSize.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# The maximum file-size for a file to be committed in kB
+max_file_size = 130.0
+
+# The path to the git-binary:
+git_executable = "@GIT_EXECUTABLE@"
+
+def sizeof_fmt(num):
+    """
+    This function will return a human-readable filesize-string
+    like "3.5 MB" for it's given 'num'-parameter.
```
Comment (kidder) [inline tools/Hooks/CheckFileSize.py]: it's -> its

File: tools/Hooks/ClangFormat.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+# The general case is needed on a Mac
+clang_format_list = ["git-clang-format",
+                     "git-clang-format-4.0",
+                     "git-clang-format-3.9",
+                     "git-clang-format-3.8"]
+
+git_executable = "@GIT_EXECUTABLE@"
+
+def which(program):
+    """ used to find the binary if it does not exist returns None """
```
Comment (kidder) [inline tools/Hooks/ClangFormat.py]: binary if  ->   binary; if

File: tools/Hooks/ClangFormat.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        break
+
+if not clang_format:
+    sys.exit(0)
+
+# extract the clang-format-a.b part of the string
+clang_format = re.search(".*\/git-(clang-format[-0-9\.]*)",
+                         clang_format).group(1)
+
+
```
Comment (kidder) [inline tools/Hooks/ClangFormat.py]: remove this blank line

File: tools/Hooks/pre-commit.sh [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+fi
+
+found_tabs=`grep --with-filename -n -P '\t' $commit_files`
+if [[ $found_tabs != "" ]]; then
+    echo "Found tabs in some files:"
+    echo "$found_tabs"
+fi
+
+no_license_found=`grep -L "^.*Distributed under the MIT License" $commit_files`
+# If commit_files is empty, then no_license_found search standard input, which
```
Comment (kidder) [inline tools/Hooks/pre-commit.sh]: search -> searches

**Review by kidder — CHANGES_REQUESTED** [review]

File: tools/Hooks/ClangFormat.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+# List of all the clang-format versions we are willing to use
+# The general case is needed on a Mac
+clang_format_list = ["git-clang-format",
+                     "git-clang-format-4.0",
+                     "git-clang-format-3.9",
+                     "git-clang-format-3.8"]
+
+git_executable = "@GIT_EXECUTABLE@"
+
+
```
Comment (kidder) [inline tools/Hooks/ClangFormat.py:20]: two blank lines

**Review by kidder — CHANGES_REQUESTED** [review]
please remove one of the blank lines

File: tools/Hooks/ClangFormat.py (hunk 5 above)
Comment (nilsdeppe) [reply tools/Hooks/ClangFormat.py:20]: This is mandated by python style guide, pep8...

Comment (nilsdeppe) [reply tools/Hooks/ClangFormat.py:20]: Here's the link: https://www.python.org/dev/peps/pep-0008/#blank-lines

**Review by kidder — APPROVED** [review]

**Pull request merged**